rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -285,6 +295,148 @@ def _decode_token_swap(self, context: DecoderContext) -> DecodingOutput:
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == user_address and
+                event.balance.amount in token_amounts
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_HOP
+                event.notes = f'Deposit {event.balance.amount} {event.asset.symbol_or_name()} to Hop'  # noqa: E501
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == user_address
+            ):
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_HOP
+                event.notes = f'Receive {event.balance.amount} {event.asset.symbol_or_name()} after providing liquidity in Hop'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_remove_liquidity(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a RemoveLiquidity event. This event is emitted when both sides of the
+        liquidity pool is withdrawn."""
+        if (liquidity_data := self._decode_common_liquidity(context)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        user_address, _, token_amounts = liquidity_data
+        lp_amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.transaction.input_data[4:36]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        return self._handle_event_decoding(
+            decoded_events=context.decoded_events,
+            user_address=user_address,
+            lp_amount=lp_amount,
+            token_amounts=token_amounts,
+        )
+
+    def _decode_remove_liquidity_one(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes a RemoveLiquidityOne event. This event is emitted when only one side of the
+        liquidity pool is withdrawn."""
+        if (liquidity_data := self._decode_common_liquidity(context)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        user_address, _, token_amounts = liquidity_data
+        lp_amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        return self._handle_event_decoding(
+            decoded_events=context.decoded_events,
+            user_address=user_address,
+            lp_amount=lp_amount,
+            token_amounts=token_amounts,
+        )
+
+    def _decode_common_liquidity(self, context: DecoderContext) -> tuple | None:
```

Comment:
properly type the tuple here. If I'm not mistaken it is

`tuple[ChecksumEvmAddress, str, set[FVal]]`